src/gui/wizard/owncloudoauthcredspage.cpp

```diff
@@ -0,0 +1,110 @@
+#include <creds/credentialsfactory.h>
+/*
+ * Copyright (C) by Krzesimir Nowak <krzesimir@endocode.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include <QVariant>
+
+#include "wizard/owncloudoauthcredspage.h"
+#include "theme.h"
+#include "account.h"
+#include "cookiejar.h"
+#include "wizard/owncloudwizardcommon.h"
+#include "wizard/owncloudwizard.h"
+#include "creds/httpcredentialsgui.h"
+
+namespace OCC
+{
+
+OwncloudOAuthCredsPage::OwncloudOAuthCredsPage()
+    : AbstractCredentialsWizardPage(),
+      _afterInitialSetup(false)
+
+{}
+
+void OwncloudOAuthCredsPage::setVisible(bool visible)
+{
+    if (!_afterInitialSetup) {
+        QWizardPage::setVisible(visible);
+        return;
+    }
+
+    if (isVisible() == visible) {
+        return;
+    }
+    if (visible) {
```

Comment:
Add a note for why this exists? I think it's resetting for when the 'Back' button is used?